core/src/main/java/org/pentaho/platform/scheduler2/quartz/QuartzScheduler.java

```diff
@@ -708,6 +745,41 @@ public List<IJob> getJobs( IJobFilter filter ) throws SchedulerException {
     return jobs;
   }
 
+  protected Date getLastRun( Trigger trigger ) {
+    Date previousTriggerNow = getPreviousTriggerNow( trigger );
+    Date previousFireTime = trigger.getPreviousFireTime();
+
+    if ( previousTriggerNow == null ) {
+      return previousFireTime;
+    }
+
+    return ( previousFireTime == null || previousTriggerNow.after( previousFireTime ) )
+         ? previousTriggerNow : previousFireTime;
+  }
+
+  private Date getPreviousTriggerNow( Trigger trigger ) {
+    JobDetail jobDetail;
+
+    try {
+      jobDetail = getQuartzScheduler().getJobDetail( trigger.getJobKey() );
+    } catch ( org.quartz.SchedulerException e ) {
+      logger.warn( "Unable to obtain previous trigger now", e );
```

Comment:
The message should be about not obtaining the job details for the given job... and it would also be a good idea to log the job key information.